bolt-contracts/src/contracts/BoltEigenLayerMiddleware.sol

```diff
@@ -0,0 +1,441 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {Ownable} from "lib/openzeppelin-contracts/contracts/access/Ownable.sol";
+import {Time} from "lib/openzeppelin-contracts/contracts/utils/types/Time.sol";
+import {EnumerableMap} from "lib/openzeppelin-contracts/contracts/utils/structs/EnumerableMap.sol";
+import {EnumerableSet} from "lib/openzeppelin-contracts/contracts/utils/structs/EnumerableSet.sol";
+import {IERC20} from "lib/openzeppelin-contracts/contracts/token/ERC20/IERC20.sol";
+
+import {MapWithTimeData} from "../lib/MapWithTimeData.sol";
+import {IBoltValidators} from "../interfaces/IBoltValidators.sol";
+import {IBoltMiddleware} from "../interfaces/IBoltMiddleware.sol";
+import {IBoltManager} from "../interfaces/IBoltManager.sol";
+
+import {IStrategyManager} from "@eigenlayer/src/contracts/interfaces/IStrategyManager.sol";
+import {IAVSDirectory} from "@eigenlayer/src/contracts/interfaces/IAVSDirectory.sol";
+import {IDelegationManager} from "@eigenlayer/src/contracts/interfaces/IDelegationManager.sol";
+import {ISignatureUtils} from "@eigenlayer/src/contracts/interfaces/ISignatureUtils.sol";
+import {IStrategy} from "@eigenlayer/src/contracts/interfaces/IStrategy.sol";
+import {AVSDirectoryStorage} from "@eigenlayer/src/contracts/core/AVSDirectoryStorage.sol";
+import {DelegationManagerStorage} from "@eigenlayer/src/contracts/core/DelegationManagerStorage.sol";
+import {StrategyManagerStorage} from "@eigenlayer/src/contracts/core/StrategyManagerStorage.sol";
+
+contract BoltEigenLayerMiddleware is IBoltMiddleware, Ownable {
+    using EnumerableSet for EnumerableSet.AddressSet;
+    using EnumerableMap for EnumerableMap.AddressToUintMap;
+    using MapWithTimeData for EnumerableMap.AddressToUintMap;
+
+    // ========= STORAGE =========
+
+    /// @notice Validators registry, where validators are registered via their
+    /// BLS pubkey and are assigned a sequence number.
+    IBoltManager public boltManager;
```

Comment:
Why does this need to point to the manager? It seems like we only need read access to `BoltValidators` which we can do directly.

Reply:
Because on the manager we can upgrade the implementation of `BoltValidators`

Reply:
Got it, given this: https://github.com/chainbound/bolt/issues/273

We may not need it since `BoltValidators` itself (as a storage contract) will not need to be upgraded necessarily.